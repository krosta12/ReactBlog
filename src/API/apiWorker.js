import axios from 'axios';

import { URL } from '../const';

const CreateUserMe = axios.create({
	baseURL: URL,
});

CreateUserMe.interceptors.request.use(
	(config) => {
		const token = localStorage.getItem('token');

		if (token) {
			config.headers.Authorization = token;
		}
		return config;
	},
	(error) => {
		return Promise.reject(error);
	}
);

CreateUserMe.interceptors.response.use((config) => {
	return config;
});

export async function post(url, object) {
	const el = await CreateUserMe.post(url, object);
	return el;
}

export async function get(url) {
	const el = await CreateUserMe.get(url);
	return el;
}

export async function _delete(url) {
	await CreateUserMe.delete(url);
}

export async function put(url, post, id) {
	await CreateUserMe.put(`${url}/${id}`, post);
}
