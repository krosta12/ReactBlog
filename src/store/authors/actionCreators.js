import { createSlice } from '@reduxjs/toolkit';

import { authorsGetter } from '../asyncAPI/ReduxAsyncRequests';

const authorSlice = createSlice({
	name: 'authorReducerSlice',
	initialState: { authorsInitialState: null },
	reducers: {
		getAllAuthors: (state, action) => {
			state.authorsInitialState = [...action.payload];
			return state;
		},
		setAuthorsToList: (state, action) => {
			state.authorsInitialState = [
				...state.authorsInitialState,
				action.payload,
			];
			return state;
		},
	},
	extraReducers: {
		[authorsGetter.fulfilled]: (state, action) => {
			try {
				state.authorsInitialState = action.payload;
			} catch (error) {
				alert(`error - ${error.status}`);
			}
		},
		[authorsGetter.rejected]: () => {
			alert('error, try later');
		},
	},
});

export const { getAllAuthors, setAuthorsToList } = authorSlice.actions;

export default authorSlice.reducer;
