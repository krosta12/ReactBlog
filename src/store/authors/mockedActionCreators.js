import { createSlice } from '@reduxjs/toolkit';

import { authorsGetter } from '../asyncAPI/ReduxAsyncRequests';

const authorSlice = createSlice({
	name: 'authorReducerSlice',
	initialState: {
		authorsInitialState: [
			{ name: '111111', id: '715de352-9124-448c-9a15-4285c1390999' },
			{ name: 'SECOND', id: 'b82afc53-804d-407c-8656-b3cdfeb986f1' },
		],
	},
	reducers: {
		getAllAuthors: (state, action) => {
			state.authorsInitialState = [...action.payload];
			return state;
		},
		setAuthorsToList: (state, action) => {
			console.log('d');
			state.authorsInitialState = [
				...state.authorsInitialState,
				action.payload,
			];
			return state;
		},
	},
	// extraReducers: {
	// 	[authorsGetter.fulfilled]: (state, action) => {
	// 		try {
	// 			state.authorsInitialState = action.payload;
	// 		} catch (error) {
	// 			alert(`error - ${error.status}`);
	// 		}
	// 	},
	// 	[authorsGetter.rejected]: () => {
	// 		alert('error, try later');
	// 	},
	// },
});

export const { getAllAuthors, setAuthorsToList } = authorSlice.actions;

export default authorSlice.reducer;
