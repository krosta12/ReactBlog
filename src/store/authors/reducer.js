export function name1() {
	alert('author1');
}
export function name2() {
	alert('author2');
}
export function name3() {
	alert('author3');
}
//я хз что с эти делать, это вырезаный кусок кода
