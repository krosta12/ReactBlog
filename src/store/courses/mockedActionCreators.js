import { createSlice } from '@reduxjs/toolkit';

const actionCreators = createSlice({
	name: 'coursesSlice',
	initialState: {
		initialList: [
			{
				title: 'NEW2322',
				description: '21132dssd2',
				duration: 213,
				authors: [
					'b82afc53-804d-407c-8656-b3cdfeb986f1',
					'715de352-9124-448c-9a15-4285c1390999',
				],
				creationDate: '29/06/2023',
				id: '12d98fbc-5789-4811-8732-64857f221155',
			},
			{
				title: 'SEc112',
				description: '1edfsgfs',
				duration: 31124,
				authors: [
					'715de352-9124-448c-9a15-4285c1390999',
					'b82afc53-804d-407c-8656-b3cdfeb986f1',
				],
				creationDate: '13/07/2023',
				id: 'cc7c5455-2836-40bd-932c-e7d9e06de0b6',
			},
		],
		// [],
	},
	reducers: {
		getAllCourses(state, action) {
			state.initialList = [...action.payload];
			return state;
		},
		setCreatedCource(state, action) {
			state.initialList = [...state.initialList, action.payload];
			return state;
		},
		clearAllCources(state) {
			state.initialList = [];
			return state;
		},
		returnInitialStateTest(state) {
			return state;
		},
	},
});

export const {
	getAllCourses,
	setCreatedCource,
	clearAllCources,
	returnInitialStateTest,
} = actionCreators.actions;

export default actionCreators.reducer;
