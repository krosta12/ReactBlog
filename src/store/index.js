const store = {
    user: {
    isAuth: false,//boolean
    name: '',//string
    email: '',//string
    token: '',//string
    },
    courses: [], 
    authors: []
   }