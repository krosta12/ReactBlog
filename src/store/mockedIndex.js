import { configureStore } from '@reduxjs/toolkit';
import mockedCourcesReducer from './courses/mockedActionCreators';
import mokedUserReducer from './user/mockedActionCreators';
import mockedAuthorReducers from './authors/mockedActionCreators';

const RootReducer = {
	mokedUserReducer,
	mockedCourcesReducer,
	mockedAuthorReducers,
};

const store = configureStore({
	reducer: RootReducer,
});

export default store;
