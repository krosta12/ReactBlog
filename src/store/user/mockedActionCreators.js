import { createSlice } from '@reduxjs/toolkit';

const userActionCreators = createSlice({
	name: 'userReducers',
	initialState: {
		user: {
			isAuth: true,
			name: 'admin',
			email: 'admin@email.com',
			token: 'token',
		},
	},
	reducers: {
		saveUser(state, action) {
			state.user.isAuth = true;
			state.user.name = action.payload.name;
			state.user.email = action.payload.email;
			state.user.token = localStorage.getItem('token');
			return state;
		},
		deleteUser(state) {
			state.user.isAuth = false;
			state.user.name = '';
			state.user.email = '';
			state.user.token = '';
			return state;
		},
	},
});

export const { saveUser, deleteUser } = userActionCreators.actions;
export default userActionCreators.reducer;
