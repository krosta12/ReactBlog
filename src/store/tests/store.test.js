import { fireEvent, render, screen } from '@testing-library/react';
import store from '../mockedIndex';
import CreateCource from '../../components/CreateCourse/CreateCourse';
import { Provider } from 'react-redux';
import userEvent from '@testing-library/user-event';
import Input from '../../common/Input/input';
import * as Redux from 'react-redux';
import mockedActionCreators, {
	getAllCourses,
	setCreatedCource,
} from '../courses/mockedActionCreators';
import { post } from '../../API/apiWorker';
import { setAuthorsToList } from '../authors/mockedActionCreators';

const mockedCources = {
	title: 'NewTitleForTest',
	description: 'NewDescriptionForTest',
	duration: 123,
	authors: ['b82afc53-804d-407c-8656-b3cdfeb986f1'],
	creationDate: '00/00/0000',
	id: '12d98fbc-5789-4811-8732-64857f221155',
};

describe('Reducers tests', () => {
	test('reducer should return the initial state', async () => {
		const initialState = store.getState();
		expect(initialState).toMatchSnapshot();
	});
	// test(`CourseForm 'Create author' click button should call dispatch`, async () => {
	// 	dispatch = jest.fn();

	// 	jest.mock('../../API/apiWorker', () => {
	// 		post: jest.fn();
	// 		get: jest.fn();
	// 	});
	// 	render(
	// 		<Provider store={store}>
	// 			<CreateCource inputAuthorName={'name'} setInputAuthorName={jest.fn()} />
	// 		</Provider>
	// 	);
	// 	const container = screen.getByText('Create author');
	// 	expect(container).toMatchSnapshot();
	// 	userEvent.dblClick(container);
	// 	expect(setAuthorsToList).toHaveBeenCalled();
	// });
	test('reducer should handle SAVE_COURSE and returns new state', async () => {
		store.dispatch(setCreatedCource(mockedCources));
		mockedActionCreators({ initialList: [] }, setCreatedCource(mockedCources));
		const saveUserHandller = store.getState();
		expect(saveUserHandller).toMatchSnapshot();
	});
	test('reducer should handle GET_COURSES and returns new state', async () => {
		store.dispatch(getAllCourses([mockedCources]));
		const saveUserHandller = store.getState();
		expect(saveUserHandller).toMatchSnapshot();
	});

	test(`CourseForm 'Add author' button click should add an author to course authors list`, async () => {
		render(
			<Provider store={store}>
				<CreateCource inputAuthorName={'name'} setInputAuthorName={jest.fn()} />
			</Provider>
		);
		const oldAuthorsList = screen.findAllByTestId('AddAuthorTest');
		expect(oldAuthorsList.length).toEqual(undefined);
		const container = screen.getAllByText('Add author');
		fireEvent.click(container[0]);
		const newAuthorList = screen.getAllByTestId('AddAuthorTest');
		expect(newAuthorList.length).toEqual(1);
	});

	test(`CourseForm 'Delete author' button click should delete an author from the course list`, () => {
		render(
			<Provider store={store}>
				<CreateCource inputAuthorName={'name'} setInputAuthorName={jest.fn()} />
			</Provider>
		);

		const addContainer = screen.getAllByText('Add author');
		fireEvent.click(addContainer[0]);
		const oldAuthorsList = screen.findAllByTestId('AddAuthorTest');
		expect(oldAuthorsList.length).toEqual(undefined);
		const deleteContainer = screen.getAllByText('Delete author');
		fireEvent.click(deleteContainer[0]);
		const newAplAuthorList = screen.findAllByText('Delete author');
		expect(newAplAuthorList.length).toEqual(undefined);
	});
});
