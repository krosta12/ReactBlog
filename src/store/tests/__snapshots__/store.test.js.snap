// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reducers tests reducer should handle GET_COURSES and returns new state 1`] = `
Object {
  "mockedAuthorReducers": Object {
    "authorsInitialState": Array [
      Object {
        "id": "715de352-9124-448c-9a15-4285c1390999",
        "name": "111111",
      },
      Object {
        "id": "b82afc53-804d-407c-8656-b3cdfeb986f1",
        "name": "SECOND",
      },
    ],
  },
  "mockedCourcesReducer": Object {
    "initialList": Array [
      Object {
        "authors": Array [
          "b82afc53-804d-407c-8656-b3cdfeb986f1",
        ],
        "creationDate": "00/00/0000",
        "description": "NewDescriptionForTest",
        "duration": 123,
        "id": "12d98fbc-5789-4811-8732-64857f221155",
        "title": "NewTitleForTest",
      },
    ],
  },
  "mokedUserReducer": Object {
    "user": Object {
      "email": "admin@email.com",
      "isAuth": true,
      "name": "admin",
      "token": "token",
    },
  },
}
`;

exports[`Reducers tests reducer should handle SAVE_COURSE and returns new state 1`] = `
Object {
  "mockedAuthorReducers": Object {
    "authorsInitialState": Array [
      Object {
        "id": "715de352-9124-448c-9a15-4285c1390999",
        "name": "111111",
      },
      Object {
        "id": "b82afc53-804d-407c-8656-b3cdfeb986f1",
        "name": "SECOND",
      },
    ],
  },
  "mockedCourcesReducer": Object {
    "initialList": Array [
      Object {
        "authors": Array [
          "b82afc53-804d-407c-8656-b3cdfeb986f1",
          "715de352-9124-448c-9a15-4285c1390999",
        ],
        "creationDate": "29/06/2023",
        "description": "21132dssd2",
        "duration": 213,
        "id": "12d98fbc-5789-4811-8732-64857f221155",
        "title": "NEW2322",
      },
      Object {
        "authors": Array [
          "715de352-9124-448c-9a15-4285c1390999",
          "b82afc53-804d-407c-8656-b3cdfeb986f1",
        ],
        "creationDate": "13/07/2023",
        "description": "1edfsgfs",
        "duration": 31124,
        "id": "cc7c5455-2836-40bd-932c-e7d9e06de0b6",
        "title": "SEc112",
      },
      Object {
        "authors": Array [
          "b82afc53-804d-407c-8656-b3cdfeb986f1",
        ],
        "creationDate": "00/00/0000",
        "description": "NewDescriptionForTest",
        "duration": 123,
        "id": "12d98fbc-5789-4811-8732-64857f221155",
        "title": "NewTitleForTest",
      },
    ],
  },
  "mokedUserReducer": Object {
    "user": Object {
      "email": "admin@email.com",
      "isAuth": true,
      "name": "admin",
      "token": "token",
    },
  },
}
`;

exports[`Reducers tests reducer should return the initial state 1`] = `
Object {
  "mockedAuthorReducers": Object {
    "authorsInitialState": Array [
      Object {
        "id": "715de352-9124-448c-9a15-4285c1390999",
        "name": "111111",
      },
      Object {
        "id": "b82afc53-804d-407c-8656-b3cdfeb986f1",
        "name": "SECOND",
      },
    ],
  },
  "mockedCourcesReducer": Object {
    "initialList": Array [
      Object {
        "authors": Array [
          "b82afc53-804d-407c-8656-b3cdfeb986f1",
          "715de352-9124-448c-9a15-4285c1390999",
        ],
        "creationDate": "29/06/2023",
        "description": "21132dssd2",
        "duration": 213,
        "id": "12d98fbc-5789-4811-8732-64857f221155",
        "title": "NEW2322",
      },
      Object {
        "authors": Array [
          "715de352-9124-448c-9a15-4285c1390999",
          "b82afc53-804d-407c-8656-b3cdfeb986f1",
        ],
        "creationDate": "13/07/2023",
        "description": "1edfsgfs",
        "duration": 31124,
        "id": "cc7c5455-2836-40bd-932c-e7d9e06de0b6",
        "title": "SEc112",
      },
    ],
  },
  "mokedUserReducer": Object {
    "user": Object {
      "email": "admin@email.com",
      "isAuth": true,
      "name": "admin",
      "token": "token",
    },
  },
}
`;
